mkosi

```diff
@@ -3956,6 +3961,9 @@ def parse_args(argv: Optional[List[str]]=None) -> Dict[str, CommandLineArguments
     if args_pre_parsed.all and args_pre_parsed.default_path:
         raise MkosiParseException("--all and --default= may not be combined.")
 
+    if args_pre_parsed.xz and args_pre_parsed.zstd:
+        raise MkosiParseException("--xz and --zstd may not be combined.")
```

Comment:
This is a completely fine solution. I don't know whether argparse groups can nest, but for this case argparse also offers `add_mutually_exclusive_group()`.

Reply:
Well i checked on that `add_mutually_exclusive_group()` but found different answers whether it works with nested groups or not.
https://bugs.python.org/issue16807
https://bugs.python.org/issue17218

and https://pymotw.com/2/argparse/#nesting-parsers

Reply:
I just checked, you can do
```
subgroup = group.add_mutually_exclusive_group()
```

Reply:
@behrmann so you want me to use `group.add_mutually_exclusive_group()` or leave as it is ?

Reply:
I guess it would make sense. Unfortunately mkosi currently targets 3.6 as baseline and I tested on 3.8. I don't know whether this works on 3.6

Reply:
python-3.8.5 here :(

Reply:
done support for add_mutually_exclusive_group() is added